### ISSUE TITLE
Question RE ACVP KAS ECC SP800-56Ar3 JSON Specification

### ISSUE THREAD

**Issue by jbarnesweb** [usnistgov/ACVP#1201]
**Protocol Section**
https://pages.nist.gov/ACVP/draft-hammett-acvp-kas-ecc-sp800-56ar3.html#name-kas-ecc-schemes
https://pages.nist.gov/ACVP/draft-hammett-acvp-kas-ecc-sp800-56ar3.html#name-generation-requirements-per
https://pages.nist.gov/ACVP/draft-hammett-acvp-kas-ecc-sp800-56ar3.html#name-supported-kas-ffcs

**Protocol Question**
How do you map ephemeralUnified, staticUnified and onePassUnified to the generation requirements (first and second link)?
Has the protocol specification conflated ECC schemes with FCC schemes in the generation requirements (second link)?
Why is there a section named Supported KAS-FFC schemes (third link) in the ECC specification?

**Comment by jbarnesweb** [comment]
I'm assuming ephemeralUnified C(2e, 0s) maps to dhEphem C(2e, 0s), but that is an FFC DH scheme according to SP 800-56Ar3.
Likewise onePassUnified may map to dhHybridOneFlow, but again that's an FFC DH scheme.
By the same token staticUnified and the FFC dhStatic have the same key agreement key characteristics, C(0e, 2s).

The confusing point is using KAS-FFC schemes to describe the generation requirements for KAS-ECC.

**Comment by Kritner** [comment]
> How do you map ephemeralUnified, staticUnified and onePassUnified to the generation requirements (first and second link)?

The table (snip):

![image](https://user-images.githubusercontent.com/8814983/125082148-dc419800-e094-11eb-8300-4307bdcd2bbf.png)

Was put together by going through the SP800-56Ar2 (at the time) and mapping out what is the "expectation" per party, per scheme as specified by the document.  It was quite a while ago, but I needed to do that mapping for our own implementation, so I made sure to document it to save some poor soul the trouble of having to do it themselves when they inevitably implement it.

The information that was pulled out of the tables in the standard was in the form of:

![image](https://user-images.githubusercontent.com/8814983/125083020-dd26f980-e095-11eb-92d5-37c52e5ced1a.png)

so just pulled that information from each individual scheme in the document.

> Why is there a section named Supported KAS-FFC schemes (third link) in the ECC specification?

Just a miss from doing the ECC and FFC documents, the FFC document was likely created first, and the ECC one closely mirrors it, so it was used as a baseline. 

> I'm assuming ephemeralUnified C(2e, 0s) maps to dhEphem C(2e, 0s), but that is an FFC DH scheme according to SP 800-56Ar3.
> Likewise onePassUnified may map to dhHybridOneFlow, but again that's an FFC DH scheme.
By the same token staticUnified and the FFC dhStatic have the same key agreement key characteristics, C(0e, 2s).
> 
> The confusing point is using KAS-FFC schemes to describe the generation requirements for KAS-ECC.

Regarding the "ephemeralUnified" -> "dhEphem", that was likely just a copy/paste error on my part when doing the ECC vs FFC specifications for the ACVP repository.

I believe the content is still accurate, would just need to update the information in this repo to have the automatically built documentation updated to reflect the correct scheme name.

I'm currently otherwise occupied with things, but will be able to get to those documentation updates at some point if no one else picks them up.

Feel free to submit a PR with the changes if you'd like.
